### ISSUE TITLE
addWeekDays() not working as expected with non-UTC timezone

### ISSUE THREAD

**Issue by chamby** [briannesbitt/Carbon#896]
The start date is already on a weekend (Saturday). I would expect both of these functions to set the date to Monday. In actuality `addWeekDays` ends up on a Sunday. `nextWeekday` ends up on a Monday. This only appears to be a problem with non-UTC timezones. See also #735 for a similar issue that I reported in the past.

**America/New_York**

```php
$date_ny = new Carbon\Carbon('2017-02-11 17:13:07.000000', 'America/New_York');
print_r($date_ny);

// Carbon\Carbon Object
// (
// 	[date] => 2017-02-11 17:13:07.000000
// 	[timezone_type] => 3
// 	[timezone] => America/New_York
// )

// INCORRECT
$one_weekday = $date_ny->copy()->addWeekDays(1);
print_r($one_weekday);

// Carbon\Carbon Object
// (
// 	[date] => 2017-02-12 17:13:07.000000
// 	[timezone_type] => 3
// 	[timezone] => America/New_York
// )

// CORRECT
$next_weekday = $date_ny->copy()->nextWeekday();
print_r($next_weekday);

// Carbon\Carbon Object
// (
// 	[date] => 2017-02-13 17:13:07.000000
// 	[timezone_type] => 3
// 	[timezone] => America/New_York
// )
```

**UTC**

```php
$date_utc = new Carbon\Carbon('2017-02-11 17:13:07.000000', 'UTC');
print_r($date_utc);

// Carbon\Carbon Object
// (
// 	[date] => 2017-02-11 17:13:07.000000
// 	[timezone_type] => 3
// 	[timezone] => UTC
// )

// CORRECT
$one_weekday = $date_utc->copy()->addWeekDays(1);
print_r($one_weekday);

// Carbon\Carbon Object
// (
// 	[date] => 2017-02-13 17:13:07.000000
// 	[timezone_type] => 3
// 	[timezone] => UTC
// )

// CORRECT
$next_weekday = $date_utc->copy()->nextWeekday();
print_r($next_weekday);

// Carbon\Carbon Object
// (
// 	[date] => 2017-02-13 17:13:07.000000
// 	[timezone_type] => 3
// 	[timezone] => America/New_York
// )
```

**Comment by Ovsyanka** [comment]
I cant reproduce this. What Carbon version you have?

I used this testcase:

```php
    public function testWeekdaysAdding()
    {
        $date_ny = new Carbon('2017-02-11 17:13:07.000000', 'America/New_York');

        $one_weekday = $date_ny->copy()->addWeekDays(1);
        $next_weekday = $date_ny->copy()->nextWeekday();
        
        $this->assertEquals($one_weekday, $next_weekday);
    }
```

**Comment by chamby** [comment]
`date_default_timezone_set('UTC');`

My PHP default timezone is `UTC`, not `America/New_York`. This issue does not seem to occur when the default is set to `America/New_York`.

I see other reported issues about the default timezone causing unexpected output #883

**Comment by chamby** [comment]
@Ovsyanka 

I forgot to answer your question. I have tried with `1.22.1` and `dev-master` (commit 926aee5ab3)

**Comment by Ovsyanka** [comment]
I added default timezone set in my supposed test (edited it) and now the issue reproduced.

**Comment by Ovsyanka** [comment]
And yes, seems like there the same reason as in #883 and my pull-request #884 fixing this.

**Comment by oceanwap** [comment]
I am also facing issue with Asia\Kolkata timezone.

**Comment by aaronhuisinga** [comment]
I'm able to reproduce this, as well. It certainly still seems to be an open issue.
I've used the nextWeekday workaround for the time being, but for other use cases that might not be an appropriate fix.

**Comment by Glavic** [comment]
This issue was solved in #1129. If not, let me know for reopening this issue.
